protocol/messenger_linkpreview_test.go

```diff
@@ -198,6 +198,19 @@ func (s *MessengerLinkPreviewsTestSuite) readAsset(filename string) []byte {
 	return b
 }
 
+func (s *MessengerLinkPreviewsTestSuite) Test_GetFavicon() {
```

Comment:
You could cover more cases in the test instead of just the simplest happy path.

Example:

- Any combination of `ico`, `svg`, and `png` links
- Link tags without `icon` or `shortcut icon`
- Other tags mixed inside `head`.